docs/content/docs/plugins/BuildChecks.md

```diff
@@ -0,0 +1,264 @@
+        it.revision = 4
+    }
+}
+```
+
+## Checks
+
+### Java version
+
+The Java version can influence the output of the Java compiler. 
+It leads to Gradle [cache misses](https://guides.gradle.org/using-build-cache/#diagnosing_cache_miss).\
+This check forces the same version for all builds.
+
+```groovy
+buildChecks {
+    javaVersion {
+        it.version = JavaVersion.VERSION_1_8
+    }   
+}
+```
+
+### Android SDK version
+
+Android build tools uses android.jar (`$ANDROID_HOME/platforms/android-<compileSdkVersion>/android.jar`).\ 
+We can specify only a version without a revision ([#117789774](https://issuetracker.google.com/issues/117789774)).
+
+```groovy
+android {
+    compileSdkVersion 29
+}
+```
+
+Different revisions lead to Gradle cache misses. This check forces the same revision:
+
+```groovy
+buildChecks {
+    androidSdk {
+        it.compileSdkVersion = 29
+        it.revision = 4
+    }
+}
+```
+
+### macOS localhost lookup
+
+On macOs `java.net.InetAddress#getLocalHost()` invocation can last up to 5 seconds instead of milliseconds.
+Gradle has a [workaround](https://github.com/gradle/gradle/pull/11134) but it works only inside Gradle's code.
+
+To diagnose the problem manually use [thoeni/inetTester](https://github.com/thoeni/inetTester).
+
+To fix the problem use this workaround:
+
+1. Find your computer's name - [Find your computer's name and network address](https://support.apple.com/en-us/guide/mac-help/find-your-computers-name-and-network-address-mchlp1177/mac)
+2. Add it to `/etc/hosts` config:
+
+```text
+127.0.0.1        localhost <your_host_name>.local
+::1              localhost <your_host_name>.local
+```
+
+3. Reboot the computer
+4. Check again by [thoeni/inetTester](https://github.com/thoeni/inetTester)
+
+See an original post [thoeni.io/post/macos-sierra-java](https://thoeni.io/post/macos-sierra-java/).
+
+This check automatically detects the issue:
+
+```groovy
+buildChecks {
+    macOSLocalhost { } // enabled by default
+}
+```
+
+### Dynamic dependency version
+
+Dynamic versions, such as "2.+", and snapshot versions force Gradle to check them on a remote server. 
+It slows down a [configuration time](https://guides.gradle.org/performance/#minimize_dynamic_and_snapshot_versions) 
+and makes build [less reproducible](https://reproducible-builds.org/). 
+
+This check forbids dynamic dependency versions.
+
+```groovy
+buildChecks {
+    dynamicDependencies { } // enabled by default
+}
+```
+
+### Gradle daemon reusage
+
+Gradle can run multiple daemons for [many reasons](https://docs.gradle.org/5.0/userguide/gradle_daemon.html#sec:why_is_there_more_than_one_daemon_process_on_my_machine).
+
+If you use `buildSrc` in the project with standalone Gradle wrapper, this check will verify common problems to reuse it.
+
+```groovy
+buildChecks {
+    gradleDaemon { } // enabled by default
+}
+```
+
+### Kotlin API dependency
+
+Gradle `api` dependencies are not working properly in kotlin plugin. 
```

Comment:
proof needed, i think we had in issue about it in kotlin tracker

Reply:
This is interesting. I couldn't reproduce it locally with 1.3.72
I'll check it in avito project.

Reply:
Yes, it works properly. See internal PR (15681).
I've removed this check completely.